src/rpp/rpp/operators/timeout.hpp

```diff
@@ -0,0 +1,172 @@
+
+            using result_type = T;
+        };
+
+        rpp::schedulers::duration                 period;
+        RPP_NO_UNIQUE_ADDRESS TFallbackObservable fallback;
+        RPP_NO_UNIQUE_ADDRESS TScheduler          scheduler;
+
+        template<rpp::constraint::decayed_type Type, rpp::details::observables::constraint::disposable_strategy DisposableStrategy, rpp::constraint::observer Observer>
+        auto lift_with_disposable_strategy(Observer&& observer) const
+        {
+            using worker_t  = rpp::schedulers::utils::get_worker_t<TScheduler>;
+            using container = typename DisposableStrategy::template add<worker_t::is_none_disposable ? 0 : 1>::disposable_container;
+
+            const auto timeout = TScheduler::now() + period;
+
+            const auto disposable = disposable_wrapper_impl<timeout_disposable<std::decay_t<Observer>, TFallbackObservable, container>>::make(std::forward<Observer>(observer), period, fallback, timeout);
+            auto       ptr        = disposable.lock();
+            ptr->get_observer_with_timeout_under_lock()->observer.set_upstream(disposable.as_weak());
+
+            const auto worker = scheduler.create_worker();
+            if constexpr (!rpp::schedulers::utils::get_worker_t<TScheduler>::is_none_disposable)
+            {
+                if (auto d = worker.get_disposable(); !d.is_disposed())
+                    disposable.add(std::move(d));
+            }
+
+            using wrapper = timeout_disposable_wrapper<std::decay_t<Observer>, TFallbackObservable, container>;
+            worker.schedule(
+                timeout,
+                [](wrapper& handler) -> rpp::schedulers::optional_delay_to {
+                    auto locked_obs_with_timeout = handler.disposable->get_observer_with_timeout_under_lock();
+                    if (TScheduler::now() < locked_obs_with_timeout->timeout)
+                        return rpp::schedulers::delay_to(locked_obs_with_timeout->timeout);
+
+                    if (!handler.disposable->is_disposed())
+                    {
+                        handler.disposable->dispose();
+                        handler.disposable->get_fallback().subscribe(std::move(locked_obs_with_timeout->observer));
+                    }
+                    return std::nullopt;
+                },
+                wrapper{ptr});
+
+            return rpp::observer<Type, timeout_observer_strategy<std::decay_t<Observer>, TFallbackObservable, container, TScheduler>>{std::move(ptr)};
+        }
+    };
+} // namespace rpp::operators::details
+
+namespace rpp::operators
+{
+    template<rpp::constraint::observable TFallbackObservable, rpp::schedulers::constraint::scheduler TScheduler /* = rpp::schedulers::immediate*/>
+    auto timeout(rpp::schedulers::duration period, TFallbackObservable&& fallback_observable, const TScheduler& scheduler /* = {}*/)
```

Comment:
Is it possible to add a variant with no fallback? And in case of timeout it just errors

Reply:
Yeah, have plans to do it, just draft :)